app/org/sagebionetworks/bridge/redis/JedisTransaction.java

```diff
@@ -0,0 +1,52 @@
+package org.sagebionetworks.bridge.redis;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.Transaction;
+import redis.clients.jedis.exceptions.JedisException;
+
+/**
+ * Provides abstraction over Jedis's transactions.
+ */
+final class JedisTransaction {
+
+    private final Logger logger = LoggerFactory.getLogger(JedisTransaction.class);
+
+    private final Jedis jedis;
+    private final Transaction transaction;
+
+    JedisTransaction(Jedis jedis) {
+        this.jedis = jedis;
+        this.transaction = jedis.multi();
+    }
+
+    public JedisTransaction setex(final String key, final int seconds, final String value) {
+        transaction.setex(key, seconds, value);
+        return this;
+    }
+
+    public List<Object> exec() {
+        try (Jedis jedis = this.jedis) {
+            return transaction.exec();
+        }
+    }
+
+    public String discard() {
+        try (Jedis jedis = this.jedis) {
+            return transaction.discard();
+        }
+    }
+
+    @Override
+    public void finalize() {
```

Comment:
finalize() is called by the garbage collector, and iirc it's generally considered an unreliable form of cleanup. I think it would be cleaner to have JedisTransaction implement Closeable and have a close method instead of a finalize() method.

Yes, this would require callers to call close (or at the very least, try-with-resources), but I think explicitly enforcing close would be cleaner than exec() and discard() implicitly calling close.